src/dataflow-types/src/plan/mod.rs

```diff
@@ -879,10 +879,30 @@ impl<T: timely::progress::Timestamp> Plan<T> {
 
     /// An empty list of arrangement keys indicates that only a `Collection` stream can
     /// be assumed to exist.
+    #[tracing::instrument(
+        level = "debug",
```

Comment:
do we expect people to want these traces by default? or do we expect it to be specific to optimization engineers debugging optimization? if its the latter, then `"trace"` is probably a better option here

Reply:
OK, although I think that at some point we should consider tracking at least the top-level optimizer spans by default if possible.

Reply:
I expect to have a high-level KPI for the optimizer team based on aggregate-level end-to-end optimization time for various classes of queries (possibly optionally grouped by user, deployment). We can then monitor this KPI across version upgrades. In theory this time should not get arbitrarily worse and overall it should always improve as the product evolves.

Reply:
I changed the default tracing level to `trace` and fixed the target to the instrumented function calls to be either `compiler` or more specifically `optimizer. To see the traces locally, one now needs the following config:

```bash
MZ_OPENTELEMETRY_ENDPOINT="http://localhost:4317" \
MZ_OPENTELEMETRY_FILTER="tokio_postgres=info,debug compiler=info,debug,trace optimizer=info,debug,trace" \
bin/materialized --reset
```